pkg/controller/controller_test.go

```diff
@@ -2202,6 +2215,21 @@ func assertServiceInstanceCredentialPropertiesStateParametersUnchanged(t *testin
 	}
 }
 
+func assertServiceInstanceCredentialOrphanMitigationSet(t *testing.T, obj runtime.Object, inProgress bool) {
```

Comment:
I would like to see this assert called in a lot more of the tests. For example, in TestReconcileBindingFailureOnFinalRetry, we should be verifying that the binding has not started orphan mitigation. Ideally, we would be verifying the correct state of `OrphanMitigationInProgress` in all of the tests that test binding reconciliation. I recommend looking at how it is done for the instance tests.

Reply:
We should definitely verify that the status correctly reflects whether orphan mitigation should be set or not in all of the tests